projects/js-packages/shared-extension-utils/src/get-block-icon-from-metadata.native.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable jsdoc/no-undefined-types */
+import { SvgXml } from '@wordpress/primitives';
+import getIconColor from './get-icon-color';
+
+/**
+ * Generate an icon as a React component from the SVG markup defined in a block.json metadata file.
+ * This prevents us from duplicating the markup in various places.
+ *
+ * Note: using an `img` tag and passing the SVG markup as a data URI doesn't allow us to
+ * dynamically set the icon color later on.
+ *
+ * @param {object} metadata - Block.json content
+ * @returns {JSX.Element} Icon component
```

Comment:
Nitpick: Since the function can return an empty string, we'd need to reflect it in the JSDoc.
```suggestion
 * @returns {JSX.Element|string} Icon component
```

Reply:
Thank you, I've committed this change, it just looks like you'll need to re-approve the PR following the commit. 🙇‍♀️